arangodb-foxx-services/src/main/services/txm/index.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2023 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+import { TxManagerImpl } from './tx-manager-impl'
+import { SubscribableTxManagerDecorator } from './subcribable-tx-manager-decorator'
+
+
+export const TxManager = new SubscribableTxManagerDecorator(new TxManagerImpl())
```

Comment:
@wajda is it really necessary to have 2 TxManager? I didn't find any place with the usage of TxManagerImpl, except here. Do you really expect that we will have another implementation of the TxManager interface? 
Furthermore, subscription is really similar to BehaviourSubject. Maybe it would be more accurate if we had only one BehaviourSubject in the TxManagerImpl, to which anyone could subscribe from just one TxManager instance.

Reply:
I'm not sure if `rxjs` is compatible with the limited Foxx environment (I see `await` is used in some places of the source code, but Foxx is strictly synchronous and async logic would likely either not work or behave unpredictably. But even if it works, how is `Subject` (not `BehaviorSubject` btw) better than `EventEmmiter` in this application? To me there is no diff in the current use case, but `events` library is provided by Foxx while `rxjs` is an external one.

Reply:
Regarding the `TxManager` interface, it has two implementations: `TxManagerImpl` and `SubscribableTxManagerDecorator`. The reason for not combining them in one is _separation of concerts_. Each class implements it's own logic and then both are cohesively connected via the _Decorator_ patern.